prime-router/metadata/fhir_mapping/hl7/resource/MessageHeader.yml

```diff
@@ -15,11 +15,12 @@ source:
   valueOf: secondary/Source
   expressionType: resource
 
+#map MSH.4
 sender:
-  valueOf: resource/MessageSender
+  valueOf: datatype/HDToOrganization
   expressionType: reference
-  vars:
-    country: MSH.17
+  generateList: true
+  specs: MSH.4
```

Comment:
This is a great call out.  I think the three options are 

- open up HDToOrganization and additional fields that can be set via vars
- create a dedicated MSH -> MessageHeaderSender.yml that would map both MSH.4 and MSH.17
- fork the library to support the ability to create extended schemas

I lean slightly towards option 1 since it would still generic and the new fields would be "opt-in"

Reply:
I agree with option 1. I have a slight concern that some resources will end up with tons of these additional var fields that only apply to specific contexts, and that it won't look that different than what we have today, but maybe that concern is unwarranted. Also even if that was the case, having one template with a bunch of one-off vars is probably still preferable to having several one-off templates that are all largely similar as would be the case with option 2. Option 3 would be nice, but I doubt it'd be easy and it's probably not super justifiable to put that effort in.

Reply:
Added this to HDToOrganization.yml